idioms/str-vs-string.md

```diff
@@ -0,0 +1,89 @@
+# Using &str instead of &String
+
+## Description
+
+It is usually recommended to prefer a `&str` type over a `&String` type for arguments.  In most cases `&str` has more flexibility.  A `&String` will coerce to a `&str` type when required -- however, the opposite is not true.  
+
+## Example
+
+Let's look at a few examples.  Consider an example where we wish to determine if a word contains three consecutive vowels.  We certainly don't need to own the string to determine this so we will take a reference.  The code might look something like this:
+
+``` rust
+fn three_vowels(word: &String) -> bool {
+    let mut vowel_count = 0;
+    for char in word.chars() {
```

Comment:
`char` is the name of a built-in type, so you should use another name (e.g. `c`) to avoid confusion.